### PR TITLE
Add README.md for `retry` package.

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

Add README.md for `retry` package.

### Additional Notes

The branch is based from https://github.com/DataDog/datadog-agent/pull/7703.

### CODE REVIEW

File: pkg/forwarder/internal/retry/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+Consider the following example where the in-memory retry queue can store up to 4 transactions.
+
+At the beginning, the retry queue in memory is empty. When the transactions `tr1`, `tr2`, `tr3` and `tr4` fail, they are added successively to the retry queue which become full. When trying to add `tr5`, the oldest transactions are removed from the retry queue in memory and serialized into the file `File 1`. The number of transactions to serialize on disk is defined as a percent of the size of the retry queue. In this example, when the retry queue is full, 50% of the transactions (in terms of payload size) are serialized on disk. 
+When adding the transactions `tr6` and `tr7`, the retry queue becomes full again and the `tr4` and `tr3` are serialized to the file `File 2`.
+![Adding transactions to the retry queue](images/Enqueue.png)
+
+##### Extracting transactions from the retry queue
+
+When extracting transactions, the first time, `tr7`, `tr6` and `tr5` are returned (Every transaction stored in memory).
+During the second call, `tr4` and `tr3` are returned and `File 2` is removed.
```
Comment (sgnn7) [inline pkg/forwarder/internal/retry/README.md:31]:
```suggestion
When those transactions are sent, `File 2` content is read and removed, sending the transactions `tr4` and `tr3`.
```

File: pkg/forwarder/internal/retry/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+During the third call, `tr2` and `tr1` are returned and `File 1` is removed.
```
Comment (sgnn7) [inline pkg/forwarder/internal/retry/README.md:32]:
```suggestion
Finally, `File 1` is read and removed, sending the `tr2` and `tr1` to the intake.
```

File: pkg/forwarder/internal/retry/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+![Extracting transactions from the retry queue](images/Extract.png)
```
Comment (sgnn7) [inline pkg/forwarder/internal/retry/README.md:34]:
```suggestion
![Removing transactions from the retry queue](images/Extract.png)
```

File: pkg/forwarder/internal/retry/README.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+#### Implementations notes
+
+* There is a single retry queue for all the endpoints.
+* The files are read and written as a whole which is efficient as few reads and writes on disk are performed.
+* At agent startup, previous files are reloaded (Unknown domains and old files are removed).
```
Comment (sgnn7) [inline pkg/forwarder/internal/retry/README.md:40]:
```suggestion
* At agent startup, previous files are reloaded. Unknown domains and old files are removed.
```

**Review by sgnn7 — CHANGES_REQUESTED** [review]
@ogaca-dd Lots of non-blocking nit suggestions but overall great work on documenting this! 

I did notice that the images don't seem to show up [when reading it](https://github.com/DataDog/datadog-agent/tree/olivierg/add-readme-for-retry/pkg/forwarder/internal/retry) so that's the only thing blocking this PR for me. It also may be good to have this doc linked from the [main forwarder README](https://github.com/DataDog/datadog-agent/blob/olivierg/add-readme-for-retry/pkg/forwarder/README.md).

**Comment by ogaca-dd** [comment]
> I did notice that the images don't seem to show up when reading it so that's the only thing blocking this PR for me. 

The link is resolved as https://github.com/DataDog/datadog-agent/blob/master/pkg/forwarder/internal/retry/images/Enqueue.png. As the PR is not yet merged, the image is not displayed.

I used the same method as for https://github.com/DataDog/datadog-agent/blob/bb61ba44b0199184f02bd8b8054b466a1dc7ac39/docs/dogstatsd/internals.md#L12.

Once the PR is merged, the image should be displayed.
Let me know if you know a better technique that still use a relative path.

File: releasenotes/notes/fixwinprocesshandleshang-bce162f5c0fee73f.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    On Windows, fixes problem in process agent wherein windows processes
```
Comment (kayayarai) [inline releasenotes/notes/fixwinprocesshandleshang-bce162f5c0fee73f.yaml:11]:
```suggestion
    On Windows, fixes problem in process agent wherein Windows processes
```

**Review by kayayarai — APPROVED** [review]
One tiny capitalization edit

File: releasenotes/notes/fixwinprocesshandleshang-bce162f5c0fee73f.yaml (hunk 5 above)
Comment (ogaca-dd) [reply releasenotes/notes/fixwinprocesshandleshang-bce162f5c0fee73f.yaml]: I did a wrong merge and this change is not part of my development.

**Comment by ogaca-dd** [comment]
Closed in favor of https://github.com/DataDog/datadog-agent/pull/7848

**Pull request closed without merging**